specification/README.md

```diff
@@ -0,0 +1,33 @@
+OpenFeature Specification
+
+## Contents
+
+- [Glossary](./glossary.md)
```

Comment:
Is there any tooling to keep the table of content synchronized and correct? Could easily become out of date and be confusing.

Reply:
I've opened: https://github.com/open-feature/spec/issues/36